### PR TITLE
45722 Create page layout header & footer

### PR DESCRIPTION
**Pull request by mubeenMastek**
Add header and footer with menus

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -5,39 +5,60 @@
...
-        </div>
-    </footer>
+    <main class="pb-3">
+        @RenderBody()
+    </main>
+    <footer class="site-footer">
+            <div class="container-fluid">
+                <div class="row justify-content-between">
+                    <div class="col-12 col-lg-auto mt-3 mb-3">
+                        <a class="footer-img" href="https://www.gov.uk/ukh">
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml:48]: Please correct the link as below.
```suggestion
                        <a class="footer-img" href="https://www.gov.uk/ukho">
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -5,39 +5,60 @@
...
-                <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target=".navbar-collapse" aria-controls="navbarSupportedContent"
+    <a class="skip-to-content-link" href="#headingLevelOne">Skip to content</a>
+    <header class="site-header">
+        <nav class="navbar navbar-expand-lg navbar-toggleable-sm navbar-light bg-white border-bottom box-shadow py-lg-0 py-2">
+            <div class="banner container-fluid justify-content-start">
+                <a class="navbar-brand" id="brandLogo" href="https://www.admiralty.co.uk/">
+                    <img class="logo-img" src="~/images/Admiralty-stacked-logo.svg" alt="Admiralty Maritime Data Solutions Home">
+                </a>
+                <div class="vertical-divider"></div>
+                <a class="site-title" id="siteTitle" asp-area="" asp-controller="Home" asp-action="Index" aria-label="Maritime Safety Information Home">Maritime Safety Information</a>
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml:21]: Does header 'Maritime Safety Information' needs to be a clickable link (<a></a>)? Not sure if there is any specific reason?

**Review by usha13529 — CHANGES_REQUESTED** [review]
Few review comments to address please.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml (hunk 2 above)
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml:21]: If user on another page, so from this link can jump to Maritime Safety Information home page. (If not needed we can remove link and will add normal text).

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml]: Changes done.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/css/nm_style.css [hunk 3]
```diff
@@ -0,0 +1,238 @@
+﻿.site-header .navbar {
```
Comment (vijendr13545) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/css/nm_style.css:1]: Is `nm` in file name **nm**_style.css means Notice to Mariners (NM)? If yes then NM is a feature/area in the application and I believe css file will be common for entire application. Please can we rename css file name to be app specific.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -5,39 +5,60 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>@ViewData["Title"] - UKHO.MaritimeSafetyInformation.Web</title>
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
-    <link rel="stylesheet" href="~/css/site.css" asp-append-version="true" />
+    <link rel="stylesheet" type="text/css" href="https://unpkg.com/%40ukho/styles@1.3.21/dist/ukho.min.css" asp-append-version="true" />
     <link rel="stylesheet" href="~/UKHO.MaritimeSafetyInformation.Web.styles.css" asp-append-version="true" />
+    <link rel="stylesheet" href="~/css/nm_style.min.css" asp-append-version="true" />
```
Comment (vijendr13545) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml:10]: Please can we consider to have just a single application css file?

**Review by vijendr13545 — COMMENTED** [review]
Looks good other than few minor comments. Thanks

**Comment by vijendr13545** [comment]
Please update favicon.ico with Admiralty icon.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -5,39 +5,60 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>@ViewData["Title"] - UKHO.MaritimeSafetyInformation.Web</title>
```
[the remark this answers is not in the document]
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml]: changes done.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -5,39 +5,60 @@
...
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
-    <link rel="stylesheet" href="~/css/site.css" asp-append-version="true" />
```
[the remark this answers is not in the document]
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml:8]: file deleted.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml (hunk 4 above)
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Shared/_Layout.cshtml]: rename scss file as per suggested

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/css/nm_style.css (hunk 3 above)
Comment (mubeenMastek) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/css/nm_style.css]: rename scss file as suggested.

**Review by usha13529 — APPROVED** [review]
LGTM.

**Review by vijendr13545 — APPROVED** [review]
Looks good to me. Thank you.

**Review by barrie-cooper — APPROVED** [review]
LGTM thanks

**Pull request merged**